plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -76,13 +78,13 @@
 
   private static AnalyticsPingManager instance;
 
+  private final String endpointUrl;
   // Preference store (should be configuration scoped) from which we get UUID, opt-in status, etc.
-  private IEclipsePreferences preferences;
-  private Display display;
-  private boolean analyticsEnabled;
+  private final IEclipsePreferences preferences;
+  private final Display display;
 
-  private ConcurrentLinkedQueue<PingEvent> pingEventQueue;
-  private Job eventFlushJob = new Job("Analytics Event Submission") {
+  private final ConcurrentLinkedQueue<PingEvent> pingEventQueue;
```

Comment:
Wait, `ConcurrentArrayQueue` is not in JDK. It's from a Jetty library. I'll keep `ConcurrentLinkedQueue`.